pkg/controller/mcmhub/mcmhub_controller.go

```diff
@@ -680,32 +682,34 @@ func (r *ReconcileSubscription) IsSubscriptionCompleted(subKey types.NamespacedN
 		return true, nil
 	}
 
-	// need to wait for managed cluster reporting back
-	// When placmentdecision doesn't have target cluster decision list, managed clusters status is empty.
-	// In this case, check the clusters list by checking the placementdecision
-	// If it's indeed empty cluster list then treat the subscription as completed.
-	managedStatus := subIns.Status.Statuses
-	if len(managedStatus) == 0 {
-		clusters, err := GetClustersByPlacement(subIns, r.Client, r.logger)
-		if err != nil {
-			return false, err
-		}
+	appsubReport := &appSubStatusV1alpha1.SubscriptionReport{}
+	if err := r.Get(context.TODO(), subKey, appsubReport); err != nil {
+		return false, err
+	}
 
-		return len(clusters) == 0, nil
+	// if there are no cluster matches or
+	// not all subscriptions are deployed/inprogress in matching clusters
+	if appsubReport.Summary.Clusters == "0" {
+		return false, nil
 	}
 
-	for cluster, cSt := range managedStatus {
-		if len(cSt.SubscriptionPackageStatus) == 0 {
-			continue
-		}
+	numClusters, err := strconv.Atoi(appsubReport.Summary.Clusters)
+	if err != nil {
+		return false, err
+	}
 
-		for pkg, pSt := range cSt.SubscriptionPackageStatus {
-			if pSt.Phase != subv1.SubscriptionSubscribed {
-				r.logger.Error(fmt.Errorf("cluster %s package %s is at status %s", cluster, pkg, pSt.Phase),
-					"subscription is not completed")
-				return false, nil
-			}
-		}
+	numDeployed, err := strconv.Atoi(appsubReport.Summary.Deployed)
+	if err != nil {
+		return false, err
+	}
+
+	numInProgress, err := strconv.Atoi(appsubReport.Summary.InProgress)
+	if err != nil {
+		return false, err
+	}
+
+	if (numDeployed + numInProgress) != numClusters {
```

Comment:
The numInProgress is already a calculated field. 
https://github.com/open-cluster-management-io/multicloud-operators-subscription/blob/e9b6cffd6be1f0573cafe2b18e2cd9438fd042c7/pkg/controller/appsubsummary/appsubsummary_controller.go#L298

So it looks we can just simply check if numInProgress >0, the subscription is not complete

Reply:
PropagationFailed > 0 or Failed > 0, we also can't proceed.

The old behaviour is when the managed cluster is Subscribed then its already considered as a count towards completion.
This calculation is to make sure inprogress can be calculated as "completed" as well to match the behaviour of the old code. Otherwise, I think we can simply check numDeployed != numClusters